src/server/index.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const { createClient } = require('./sproutVideo');
+const { ELEMENT_STATE } = require('../shared');
+
+async function afterSave(asset, { config: { tce } }) {
+  const { videoId, playable, fileName, error, status } = asset.data;
+  if (!fileName || playable) return asset;
+  const { sproutVideoApiKey: apiKey } = tce;
+  const client = createClient({ apiKey });
+  const isEmptyOrError = error || !videoId || status !== ELEMENT_STATE.UPLOADED;
+  if (!isEmptyOrError) startPollingPlayableStatus(asset, client);
+  const { token } = await client.videos.getDelegatedToken();
+  asset.data.token = token;
+  asset.data.uploadUrl = client.videos.getUploadUrl();
+  return asset;
+}
+
+async function startPollingPlayableStatus(asset, client) {
+  const { videoId } = asset.data;
+  const video = await client.videos.get(videoId);
+  const isPlayable = video.state === 'deployed';
+  if (!isPlayable) return setTimeout(() => startPollingPlayableStatus(asset, client), 5000);
+  delete asset.data.token;
+  delete asset.data.uploadUrl;
+  asset.update({ data: { ...asset.data, playable: true } });
```

Comment:
We can use `lodash/omit` to avoid mutation:
```suggestion
  asset.update({
    data: { ...omit(asset.data, ['token', 'uploadUrl']), playable: true }
  });
```